src/nrnpython/nrnpy_nrn.cpp

```diff
@@ -1493,29 +1493,28 @@ static PyObject* NPySecObj_connect_safe(NPySecObj* self, PyObject* args) {
 static PyObject* NPySecObj_insert(NPySecObj* self, PyObject* args) {
     CHECK_SEC_INVALID(self->sec_);
     char* tname;
-    PyObject *tpyobj, *tpyobj2;
     if (!PyArg_ParseTuple(args, "s", &tname)) {
         PyErr_Clear();
         // if called with an object that has an insert method, use that
+        PyObject* tpyobj;
         if (PyArg_ParseTuple(args, "O", &tpyobj)) {
```

Comment:
I wonder if it makes sense to [nb::tuple](https://nanobind.readthedocs.io/en/latest/api_core.html#_CPPv4N8nanobind5tupleE) to make the error handling easier

Reply:
It would be nice, PyArg_ParseTuple` is used everywhere to parse args into the right C type. But do you know how to express the same in nanobind?

Reply:
> PyArg_ParseTuple is used everywhere 

I agree, I will see if I can cook something up, I will get back to you.

Reply:
I will leave the suggestion to yet another later PR where we try to improve the many places using PyArg_ParseTuple